test/lib/tasks/seeds_test.rb

```diff
@@ -0,0 +1,45 @@
+require 'test_helper'
+
+class SeedsTest < ActiveSupport::TestCase
+  # Disable AR transactional fixtures as we use DatabaseCleaner's truncation
+  # to empty the DB of fixtures for testing the seed script
+
+  setup do
+    self.use_transactional_fixtures = false
+    DatabaseCleaner.clean_with :truncation
+    Setting.stubs(:[]).with(:administrator).returns("root@localhost")
+    Setting.stubs(:[]).with(:send_welcome_email).returns(false)
+    Katello::Repository.stubs(:ensure_sync_notification)
+  end
+
+  def seed
+    # Authorisation is disabled usually when run from a rake db:* task
+    User.current = FactoryGirl.build(:user, :admin => true)
+    load File.expand_path("#{Rails.root}/db/seeds.rb", __FILE__)
+  end
+
+  teardown do
+    self.use_transactional_fixtures = true
+    User.current = nil
+  end
+
+  test "don't create a default location if no locations exist" do
+    Location.stubs(:exists?).returns(false)
+    seed
+    refute Location.default_location.present?
+  end
+
+  test "create a default location on seed on a fresh install" do
+    with_env('SEED_LOCATION' => 'seed_test') do
+      seed
+    end
+    assert Location.default_location.present?
+    assert_equal "seed_test", Location.default_location.name
+  end
+
+  test "setup org bindings for every org in foreman" do
+    org = Organization.create!(:name => "myOrg")
+    ForemanTasks.expects(:sync_task).with(::Actions::Katello::Organization::Create, org)
+    seed
+  end
+end
```

Comment:
Might as well add tests for the other seeds to ensure provisioning templates are configured properly, permissions are added, mail notifications are set up etc.